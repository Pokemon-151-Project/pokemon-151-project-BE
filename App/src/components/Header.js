// Contains page title, author's name, helpful links, and dropdown for user display options
import React from "react";
import Dropdown from "./Dropdown";
import useDarkMode from "../hooks/useDarkMode";
//darkmode toggle icons
import { BsMoon, BsSun } from "react-icons/bs";
//linksArray is the list of hrefs and display test to put in the nav. Stored in utils folder
import linksArray from "../utils/Headerutils/linksArray";
//generates links for the mav using the info in linksArray
import linkMaker from "../utils/Headerutils/linkMaker";

const Header = (props) => {
  const [isDarkMode, setDarkMode] = useDarkMode();

  const { changeDisplay, isShiny, setIsShiny } = props;
  return (
    <header>
      <h1 className="header-item header-h1">Pokemon 151 Project</h1>

      {/* Begin display options dropdown. See Dropdown.js */}
      <section className="darkmode-shiny-dropdown">
        <menu className="darkmode-shiny-section">
          <div className="darkmode-section">
            <div className="logo" data-testid="logo">
              {isDarkMode ? "Dark Mode" : "Light Mode"}
            </div>
            <button
              className="toggle_btn"
              data-testid="toggle_btn"
              onClick={() => {
                setDarkMode(!isDarkMode);
              }}
            >
              {isDarkMode ? (
                <BsSun color="#ff0" size="24" title="Switch to light mode" />
              ) : (
                <BsMoon size="24" title="Switch to dark mode" />
              )}
            </button>
          </div>

          {/* This is the styled big green line running through the header */}
          <div className="vertical-line" />

          {/* This button toggles whether the app shows the Shiny forms of the pokemon */}
          <div className="shiny-button-div">
            <button
              className="shiny-button"
              onClick={() => {
                setIsShiny(!isShiny);
              }}
            >
              {isShiny ? "Show Non-Shiny Forms" : "Show Shiny Forms"}
            </button>
          </div>
        </menu>

        {/* The dropdown section lives here */}
        <Dropdown changeDisplay={changeDisplay} />
      </section>

      <section className="adam-info header-item">
        <h2>Author: Adam Hinton</h2>

        {/* the linkMaker function produces header links. Update linksArray (imported above) to add/change/remove links*/}
        <nav className="header-nav">{linkMaker(linksArray)}</nav>
      </section>
    </header>
  );
};

export default Header;
