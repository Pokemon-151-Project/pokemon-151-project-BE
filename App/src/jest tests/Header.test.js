import Header from "../components/Header";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";

test("[1] Component renders without errors", () => {
  render(<Header />);
});

test("[2] Renders the expected segments of the header", () => {
  render(<Header />);
  const h1 = screen.queryByText(/pokemon 151 project/i);
  const author = screen.queryByText(/author: adam hinton/i);
  const source = screen.queryByText(/author/i);
  const dropbtn = screen.queryByText(/display options/i);

  expect(author).toBeInTheDocument();
  expect(h1).toBeInTheDocument();
  expect(source).toBeInTheDocument();
  expect(dropbtn).toBeVisible();
});

//I tried to test that the dropdown only appears when you click on the Display Options button, but the testing said the dropdown was always visible no matter what. I tried document.queryselector, screen.getByTestID, and others. Not sure what I was doing wrong.

test("[3] 'Dark Mode' text changes to 'Light Mode' and vice versa when you click dark mode toggle logo", async () => {
  render(<Header />);
  const darkModeToggleLogo = screen.getByTestId("toggle_btn");

  expect(screen.queryByText(/dark mode/i)).not.toBeVisible();
  expect(screen.queryByText(/light mode/i)).toBeVisible();

  await userEvent.click(darkModeToggleLogo);

  expect(screen.queryByText(/dark mode/i)).toBeVisible();
  expect(screen.queryByText(/light mode/i)).not.toBeVisible();
});

// This has saved a snapshot of Header.js from 5.8.2022. If anything in App changes this test will fail
//Then I will either need to update the screenshot with the new changes, or fix what error made the test fail.
//If the change is intended, run jest -u in terminal to update snapshot.
test("[4] Matches snapshot so nothing changes inadvertently", () => {
  const component = renderer.create(<Header />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
