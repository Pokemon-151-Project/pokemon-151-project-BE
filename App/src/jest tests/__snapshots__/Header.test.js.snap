// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[4] Matches snapshot so nothing changes inadvertently 1`] = `
<header>
  <h1
    className="header-item header-h1"
  >
    Pokemon 151 Project
  </h1>
  <section
    className="darkmode-shiny-dropdown"
  >
    <menu
      className="darkmode-shiny-section"
    >
      <div
        className="darkmode-section"
      >
        <div
          className="logo"
          data-testid="logo"
        >
          Dark Mode
        </div>
        <button
          className="toggle_btn"
          data-testid="toggle_btn"
          onClick={[Function]}
        >
          <svg
            color="#ff0"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "#ff0",
              }
            }
            viewBox="0 0 16 16"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Switch to light mode
            </title>
            <path
              d="M8 11a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0 1a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"
            />
          </svg>
        </button>
      </div>
      <div
        className="vertical-line"
      />
      <div
        className="shiny-button-div"
      >
        <button
          className="shiny-button"
          onClick={[Function]}
        >
          Show Shiny Forms
        </button>
      </div>
    </menu>
    <section
      className="dropdown"
    >
      <button
        className="dropbtn"
        data-testid="dropbtn"
      >
        Display Options (check this out!)
      </button>
      <menu
        className="dropdown-content"
      >
        <div>
          <div>
            <button
              className="dropbtn-inner"
              data-testid="reset-btn"
              id="reset-btn"
              onClick={[Function]}
            >
              Reset All
            </button>
          </div>
          <h5
            className="dropdown-display-title"
            data-testid="dropdown-display-title"
          >
            Pick a type:
          </h5>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="normal-btn"
            onClick={[Function]}
          >
            Normal
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="fire-btn"
            onClick={[Function]}
          >
            Fire
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="water-btn"
            onClick={[Function]}
          >
            Water
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="electric-btn"
            onClick={[Function]}
          >
            Electric
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="grass-btn"
            onClick={[Function]}
          >
            Grass
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="ice-btn"
            onClick={[Function]}
          >
            Ice
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="fighting-btn"
            onClick={[Function]}
          >
            Fighting
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="poison-btn"
            onClick={[Function]}
          >
            Poison
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="ground-btn"
            onClick={[Function]}
          >
            Ground
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="flying-btn"
            onClick={[Function]}
          >
            Flying
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="psychic-btn"
            onClick={[Function]}
          >
            Psychic
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="bug-btn"
            onClick={[Function]}
          >
            Bug
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="rock-btn"
            onClick={[Function]}
          >
            Rock
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="ghost-btn"
            onClick={[Function]}
          >
            Ghost
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="dragon-btn"
            onClick={[Function]}
          >
            Dragon
          </button>
        </div>
        <div>
          <h5
            className="dropdown-display-title"
          >
            Pick a weight:
          </h5>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="small-btn"
            onClick={[Function]}
          >
            Small
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="big-btn"
            onClick={[Function]}
          >
            Big
          </button>
        </div>
        <div>
          <h5
            className="dropdown-display-title"
          >
            Pick a height:
          </h5>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="short-btn"
            onClick={[Function]}
          >
            Short
          </button>
          <button
            className="dropbtn-inner"
            data-testid="dropbtn-inner"
            id="tall-btn"
            onClick={[Function]}
          >
            Tall
          </button>
        </div>
      </menu>
    </section>
  </section>
  <section
    className="adam-info header-item"
  >
    <h2>
      Author: Adam Hinton
    </h2>
    <nav
      className="header-nav"
    >
      <a
        href="https://github.com/adamhinton/pokemon-project"
        rel="noreferrer"
        target="_blank"
      >
        Source
      </a>
      <a
        href="https://bulbapedia.bulbagarden.net"
        rel="noreferrer"
        target="_blank"
      >
        Bulbapedia
      </a>
      <a
        href="https://pokeapi.co"
        rel="noreferrer"
        target="_blank"
      >
        PokeAPI
      </a>
      <a
        href="https://bulbapedia.bulbagarden.net/wiki/Shiny_Pok%C3%A9mon"
        rel="noreferrer"
        target="_blank"
      >
        Shinies
      </a>
    </nav>
  </section>
</header>
`;
